app/controllers/cfp/events_controller.rb

```diff
@@ -83,6 +83,17 @@ def confirm
     end
   end
 
+  def join
+    @token = params[:token] || ''
+    @event = @token.blank? ? nil : Event.find_by(invite_token: @token)
+    if request.post?
+      return redirect_to cfp_join_event_path, flash: { error: t('cfp.join_token_unknown', token: @token) } if @event.nil?
```

Comment:
It's quite complex logic. I think it's better to split it
```ruby
def join
 return unless request.post?
 @token = params[:token] || ''
 @event = @token.blank? ? nil : Event.find_by(invite_token: @token)
redirect_on_invalid_event
redirect_on_already_created
@event.event_people << EventPerson.new(person: current_user.person, event_role: 'speaker')
redirect_to edit_cfp_event_path(@event), notice: t('cfp.join_success')
end

private

def redirect_on_invalid_event
    return redirect_to cfp_join_event_path, flash: { error: t('cfp.join_token_unknown', token: @token) } if @event.nil?
 end

def redirect_on_already_created
     return redirect_to edit_cfp_event_path(@event), notice: t('cfp.join_token_already_used') if @event.people.exists?(current_user.person.id)
end

Reply:
The returns from redirect_on_invalid_event and redirect_on_already_created don't propagate up to the join function, so this does not work.